### ISSUE TITLE
NCF Recommendation training cannot converge when set the -layers parameter as 256,128,64

### ISSUE THREAD

**Issue by lingjiew** [tensorflow/models#4865]
Hi， 
I have tried the commit version eb4f6f5 of mlperf for a small model with layer parameter as 256,128,64, and it converged within 7 epochs.
But when I set the same -layers params in this project, the training cannot converged even in 20 epochs. 
Why the training is so different between this project and mlperf?

Here are my training params.
tensorflow/models:
python ncf_main.py -dataset ml-20m -hooks "" -hr_threshold 0.9562 -train_epochs 100 -learning_rate 0.0005 -batch_size 16384 -layers 256,128,64 -num_factors 64

mlperf:
python ncf.py ml-20m -l 0.0005 -b 16384 --layers 256 128 64 -f 64 --seed 1 --threshold 0.9562

**Comment by lingjiew** [comment]
Thank you for your reply! Do you plan to improve the accuracy in short term?

**Comment by robieta** [comment]
Yes, I've been working on a large scale rewrite of much of the NCF data pipeline that should be ready soon and help resolve the TF and PyTorch differences.

**Comment by lingjiew** [comment]
OK! Thank you very much!
